bucket/qbittorrent.json

```
@@ -16,14 +16,24 @@
             "hash": "01487a0e2594a5065e4d780eb012dcd0dafadc218d1b6aba69528bd6ede6afb5"
         }
     },
-    "pre_install": "Remove-Item \"$dir\\`$PLUGINSDIR\" -Force -Recurse",
+    "pre_install": "Remove-Item \"$dir\\`$PLUGINSDIR\", \"$dir\\uninst.exe\" -Force -Recurse",
```

```suggestion
    "pre_install": [
        "Remove-Item \"$dir\\`$PLUGINSDIR\" -Force -Recurse",
        "if ( (!(Test-Path \"$persist_dir\\profile\")) -and (Test-Path \"$env:APPDATA\\qBittorrent\") ) {",
        "    Write-Host \"Scoop is migrating qbittorrent to use portable mode by default.\" -ForeGroundColor Yellow",
        "    Write-Host \"For details, see: https://github.com/ScoopInstaller/Extras/pull/7173\" -ForeGroundColor Yellow",
        "    Write-Host \"Copying files from APPDATA to Scoop directory...\" -ForeGroundColor Yellow",
        "    New-Item \"$persist_dir\\profile\\config\", \"$persist_dir\\profile\\data\" -ItemType Directory | Out-Null",
        "    Copy-Item \"$env:APPDATA\\qBittorrent\\*\" \"$persist_dir\\profile\\config\" -Force | Out-Null",
        "    Copy-Item \"$env:LOCALAPPDATA\\qBittorrent\\*\" \"$persist_dir\\profile\\data\" -Force | Out-Null",
        "}"
    ],
```
`uninst.exe` does not exist in *portable* version.

The migration script should be in *pre_install*. Otherwise `$persist_dir\profile` will be created before running the script. (sorry I was wrong)

---

Sorry for the late response, you are totally right sir.